qiskit_ibm_runtime/runtime_job.py

```diff
@@ -320,12 +323,42 @@ def _set_error_message(self, job_response: Dict) -> None:
         """
         if self._status == JobStatus.ERROR:
             job_result_raw = self._api_client.job_results(job_id=self.job_id)
-            self._error_message = API_TO_JOB_ERROR_MESSAGE[
-                job_response["status"].upper()
-            ].format(self.job_id, job_result_raw)
+            self._error_message = self._error_msg_from_job_response(
+                job_response
+            ).format(self.job_id, job_result_raw)
```

Comment:
I would find it cleaner to remove the `.format(...)` part and include this in the `_error_msg_from_job_response` function. Previously where we relied on a simple mapping this was not possible but the dedicated function you introduced allows for that flexibility.